This patch adds null addr check for ioremap_prot return addr.
This fixes kernel crash seen in CM-934.

Failure was seen when ioremap_prot fails with the below
error in arch/powerpc/mm/pgtable_32.c

Jul 23 15:02:04 act-5652-24 kernel: __ioremap(): phys addr 0x7f46000 is RAM lr c00af1f4

diff --git a/mm/memory.c b/mm/memory.c
index 483e665..7f26814 100644
--- a/mm/memory.c
+++ b/mm/memory.c
@@ -3823,6 +3823,9 @@ int generic_access_phys(struct vm_area_struct *vma, unsigned long addr,
 		return -EINVAL;
 
 	maddr = ioremap_prot(phys_addr, PAGE_SIZE, prot);
+	if (!maddr)
+		return -EFAULT;
+
 	if (write)
 		memcpy_toio(maddr + offset, buf, len);
 	else
